hera_cal/utils.py

```diff
@@ -497,80 +502,94 @@ def JD2LST(JD, longitude=21.42830):
         _array = False
         JD = [JD]
 
-    # iterate over JD
-    LST = []
-    for jd in JD:
-        # construct astropy Time object
-        t = Time(jd, format='jd', scale='utc')
-        # get LST in radians at epoch of jd
-        LST.append(t.sidereal_time('apparent', longitude=longitude * unt.deg).radian)
-    LST = np.array(LST)
+    # use pyuvdata
+    LST = uvutils.get_lst_for_time(np.array(JD), latitude, longitude, altitude)
 
     if _array:
         return LST
     else:
         return LST[0]
 
 
-def LST2JD(LST, start_jd, longitude=21.42830):
+def LST2JD(LST, start_jd, allow_other_jd=False, lst_branch_cut=0.0, latitude=-30.721526120689507,
+           longitude=21.428303826863015, altitude=1051.690000018105):
     """
     Convert Local Apparent Sidereal Time -> Julian Date via a linear fit
     at the 'start_JD' anchor point.
 
     Input:
     ------
-    LST : type=float, local apparent sidereal time [radians]
+    LST : type=float or array-like, local apparent sidereal time [radians]
 
     start_jd : type=int, integer julian day to use as starting point for LST2JD conversion
+    
+    allow_other_jd : type=bool, ensure that the lst_branch_cut falls on start_jd but allow 
+                     LSTs to correspond to previous or subsequent days if necessary
+        
+    lst_branch_cut : type=float, LST that must fall during start_jd even when allow_other_jd 
+                     is True. Used as the starting point starting point where LSTs below this
+                     value map to later JDs than this LST [radians]
+    
+    latitude : type=float, degrees North of observer, default=HERA latitude
 
     longitude : type=float, degrees East of observer, default=HERA longitude
 
+    altitude : type=float, altitude in meters, default=HERA altitude
+
     Output:
     -------
-    JD : type=float, Julian Date(s). accurate to ~1 milliseconds
+    JD : type=type(LST), Julian Date(s). accurate to ~1 milliseconds
     """
     # get LST type
     if isinstance(LST, list) or isinstance(LST, np.ndarray):
+        LST = np.array(LST)
         _array = True
     else:
-        LST = [LST]
+        LST = np.array([LST])
         _array = False
 
-    # get start_JD
-    base_jd = float(start_jd)
-
-    # iterate over LST
-    jd_array = []
-    for lst in LST:
+    # increase LSTs so no values fall below the branch cut (avoiding wraps in interpolation)
+    while np.any(LST < lst_branch_cut):
+        LST[LST < lst_branch_cut] += 2 * np.pi  
```

Comment:
are you sure this doesn't affect the input LST array? Perhaps we should make a copy of it at the top just in case

Reply:
also I don't think a while is needed

Reply:
Yes, because of the line `LST = np.array(LST)` or `LST = np.array([LST])` which was `copy=True` by default.